apps/extension/src/core/views/PermissionRequest.tsx

```diff
@@ -96,7 +94,7 @@ function TransactionPermission({ transaction }: { transaction: Transaction }) {
       </Accordion>
       <Dropdown
         choices={["ask", "allow"]}
-        onSelect={async (permission) =>
+        onSelect={async (permission: "ask" | "allow") =>
```

Comment:
nit - This type can be inferred by doing `choices={["ask", "allow"] as const}`, and use a generic type in Dropdown for choices (which get passed into onSelect as values

Reply:
Something like this: 
```tsx
      <Dropdown
        choices={["ask", "allow"] as const}
```

In `Dropdown.tsx`:

```tsx
export function Dropdown<T extends string, L extends Readonly<Array<T>>>({
  styled = false,
  children,
  choices,
  onSelect
}: {
  styled?: boolean
  children: React.ReactNode
  choices: L
  onSelect: (choice: L[number]) => void
}) {
...
```

Reply:
Woah, so weird! I only had to make it a readonly array and a const, and now it works:
<img width="365" alt="Screenshot 2023-04-29 at 2 32 55 PM" src="https://user-images.githubusercontent.com/1011391/235318672-7f96a7f3-fcf1-480b-9300-ac91147d4e54.png">

Reply:
Nice find